core/domain/translation_domain.py

```diff
@@ -18,9 +18,280 @@
 
 from __future__ import annotations
 
+import enum
+
+from core import feconf
 from core import utils
 
-from typing import Dict
+from typing import Dict, List
+from typing_extensions import TypedDict
+
+
+class TranslatableContentFormat(enum.Enum):
+    """Represents all possible data types for any translatable content."""
+
+    HTML = 'html'
+    UNICODE_STRING = 'unicode'
+    SET_OF_NORMALIZED_STRING = 'set_of_normalized_string'
+    SET_OF_UNICODE_STRING = 'set_of_unicode_string'
+
+
+class TranslatableContentDict(TypedDict):
+    """Dictionary representing TranslatableContent object."""
+
+    content_id: str
+    content_value: feconf.ContentValueType
+    content_type: TranslatableContentFormat
+
+
+class TranslatableContent:
+    """TranslatableContent represents a content of a translatable object which
+    can be translated into multiple languages.
+
+    Args:
+        content_id: str. The id of the corresponding translatable content value.
+        content_value: ContentValueType. The content value which can be
+            translated.
+        content_type: TranslatableContentFormat. The type of the
+            corresponding content value.
+    """
+
+    def __init__(
+        self,
+        content_id: str,
+        content_value: feconf.ContentValueType,
+        content_type: TranslatableContentFormat
+    ) -> None:
+        self.content_id = content_id
+        self.content_value = content_value
+        self.content_type = content_type
+
+    @classmethod
+    def from_dict(
+        cls,
+        translatable_content_dict: TranslatableContentDict
+    ) -> TranslatableContent:
+        """Returns a TranslatableContent object from its dict representation.
+
+        Args:
+            translatable_content_dict: dict. Dict representation of
+                TranslatableContent object.
+
+        Returns:
+            TranslatableContent. An instance of TranslatableContent class.
+        """
+        return cls(
+            translatable_content_dict['content_id'],
+            translatable_content_dict['content_value'],
+            translatable_content_dict['content_type'])
+
+    def to_dict(self) -> TranslatableContentDict:
+        """Returns the dict representation of TranslatableContent object.
+
+        Returns:
+            TranslatableContentDict. A dict, mapping content_id, content_value
+            and content_type of a TranslatableContent instance to
+            corresponding keys 'content_id', 'content_value' and
+            'content_type'.
+        """
+        return {
+            'content_id': self.content_id,
+            'content_value': self.content_value,
+            'content_type': self.content_type
+        }
+
+
+class TranslatedContent:
+    """Represents the translated content of the TranslatableContent object.
+
+    Args:
+        content_value: ContentValueType. Represents already-translated
```

Comment:
I still can't quite be certain, from reading this description, whether this is the content that's being translated or the actual translation. Maybe you can explain this further in line 105 since that is where the term "translated content" is introduced, and then that's the definition that the reader can hold in their mind while reading the rest of the code.

Reply:
Updated the docstring.